client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcVendorCompatibility.java

```diff
@@ -88,16 +88,34 @@ public void setupServer()
         Logging.initialize();
         log = Logger.get(TestJdbcVendorCompatibility.class);
         server = TestingTrinoServer.create();
-        referenceDrivers = ImmutableList.of(new PostgresqlReferenceDriver(), new OracleReferenceDriver());
+
+        // Capture resources as soon as they are allocated. Ensure all allocate resources are cleaned up even if e.g. the last one fails to start.
+        referenceDrivers = new ArrayList<>();
+        referenceDrivers.add(new PostgresqlReferenceDriver());
+        referenceDrivers.add(new OracleReferenceDriver());
```

Comment:
i don't think having `closer` field in test class helps anything, since we still want to clear the references, so each field requires some cleanup code anyway